cmd/preflight/cmd/check_container.go

```diff
@@ -95,14 +98,39 @@ func checkContainerRunE(cmd *cobra.Command, args []string, runpreflight runPrefl
 	}
 	logger.Info("certification library version", "version", version.Version.String())
 
-	containerImage := args[0]
-
 	// Render the Viper configuration as a runtime.Config
 	cfg, err := runtime.NewConfigFrom(*viper.Instance())
 	if err != nil {
 		return fmt.Errorf("invalid configuration: %w", err)
 	}
 
+	containerImage := args[0]
+	containerImagePlatforms := []string{cfg.Platform}
+
+	manifestBytes, err := crane.Manifest(containerImage)
+	if err != nil {
+		return fmt.Errorf("could not retrieve manifest: %w", err)
+	}
+	manifest, err := cranev1.ParseManifest(bytes.NewBuffer(manifestBytes))
+	if err != nil {
+		return fmt.Errorf("could not parse manifest: %w", err)
+	}
+
+	platformChanged := cmd.Flags().Lookup("platform").Changed
+	if manifest.MediaType == types.DockerManifestList && !platformChanged {
+		manifest, err := cranev1.ParseIndexManifest(bytes.NewBuffer(manifestBytes))
```

Comment:
It feels a bit odd to have this logic here. Up until now, the manipulation of actual image metadata (i.e., everything beyond this is the image the user wanted to pass to preflight) is done in that engine.

To that end, this logic being here makes it not present in our container parsing library. I'm thinking we should internally determine whether or not were intending to parse a manifest list there instead of here so that the behavior is consistent.